components/org.wso2.carbon.identity.cors.service/src/main/java/org/wso2/carbon/identity/cors/service/internal/CORSServiceComponent.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.cors.service.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.identity.configuration.mgt.core.ConfigurationManager;
+import org.wso2.carbon.identity.cors.service.CORSManager;
+import org.wso2.carbon.identity.cors.service.internal.impl.CORSManagerImpl;
+
+/**
+ * Service component class for CORS-Service.
+ */
+@Component(name = "org.wso2.carbon.identity.cors.service", immediate = true)
+public class CORSServiceComponent {
+
+    private static final Log log = LogFactory.getLog(CORSServiceComponent.class);
+
+    /**
+     * Activate the CORSServiceComponent.
+     *
+     * @param context
+     */
+    @Activate
+    protected void activate(ComponentContext context) {
+
+        context.getBundleContext().registerService(CORSManager.class, CORSManagerImpl.getInstance(), null);
```

Comment:
let's surround with a try-catch(Throwable e) block to avoid inconvenient situations during server startup.
similar to https://github.com/wso2/carbon-identity-framework/blob/55ec4664e0a89191a7f187ed08a16b20d998273f/components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/internal/UserMgtDSComponent.java#L181